tensorflow/compiler/jit/xla_cluster_util.cc

```diff
@@ -386,4 +388,19 @@ XlaAutoClusteringSummary GetXlaAutoClusteringSummary(const Graph& graph) {
 
   return result;
 }
+
+// Implement and register a callback for querying GlobalJitLevel.
+OptimizerOptions::GlobalJitLevel GlobalJitLevelGetter(
+    OptimizerOptions::GlobalJitLevel jit_level_in_session_opts) {
+  XlaGlobalJitLevel xla_global_jit_level =
+      GetXlaGlobalJitLevel(jit_level_in_session_opts);
+  // Take the general flag to avoid the dependency on Tensorflow::Graph.
```

Comment:
I simply ignore the single_graph flags, as the graph is not accessible in `InitializeOptimizers()`, where grappler configs its optimizers.

It makes sense to me that we simply expose the general flag (,although both of the general and single_gpu flags are currently the same). Let me know what you think about this though.

Reply:
I'm worried that this makes `XlaConfigRegistry::GetGlobalJitLevel` "look" misleading.

How about this:  we make the callback return an instance of `XlaGlobalJitLevel`.  Then in optimizers/meta_optimizer.cc we do:

```
XlaGlobalJitLevel xla_global_jit_level = ...;
if (xla_global_jit_level.general is ON) {
  Disable memory opt
}
// We don't care about single GPU because the regression happens only on multi-GPU 
```

This is a bit of a hack because if the graph happens to be single GPU but the general auto-jit is enabled then we're still going to disable the optimization but I think for getting us started this is fine.  Behaviorally this is identical to what you have but I think it is clearer.

Also a bit more context: if only single_gpu is enabled we absolutely don't want any behavior change on multi-GPU graphs.  We're using `--tf_xla_auto_jit=single-gpu(2)` to stage the rollout of XLA and we want `--tf_xla_auto_jit=single-gpu(2)` to only affect single GPU graphs.

Reply:
I see your point. Thanks for the good feedback.

I made some changes:
1. Made the callback return `XlaGlobalJitLevel`, as this way is indeed less ambiguous.
2. I added check in the `MetaOptimizer` such that the memory optimizer is turned off only when JIT is on for _both_ single-gpu and multi-gpu graphs. This is a conservative approach and should be good enough for now.

Thanks for the context about how the single-gpu flag is used. It was not clear to me and it looks good to me.

Note that the swap-in/out latency issue should also exist in single gpu since the copy stream is different from device compute stream in the default Tensorflow setting.